### PR TITLE
ci: upgrade some actions related to planned GH deprecations

### PR DESCRIPTION
**Pull request by derberg**
### actions bumps forced by [node12 -> node16](https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/):

- Upgrade checkout action to `actions/checkout@v3`. It is safe as they bumped from v2 to v3 only to upgrade the node runner to node 16. No breaking changes
- Upgrade script action to `actions/github-script@v6`. There were [breaking changes in v5](https://github.com/actions/github-script#breaking-changes-in-v5) only, v6 was just node engine update. Only one workflow needed a refactor to `github.rest`
- Upgrade to latest `pascalgn/automerge-action`. No breaking changes, just for node engine deprecation
- Upgrade to latest `hmarr/auto-approve-action`. No breaking changes, just for node engine deprecation
- Upgrade docker actions from `if-docker-pr-testing.yml` workflow. No breaking changes, just for node engine deprecation
- Upgrade go actions from `if-go-pr-testing.yml` workflow. No breaking changes, just for node engine deprecation
- Upgrade `actions/setup-node` to v3. Node engine related. There is one breaking change that do not relate to us, to use `node-version` input instead of `version` - but that was always done
- Upgrade to latest `peter-evans/create-pull-request`. There are some breaking changes related to `add-paths` but we are no using it. We need to upgrade because of node enginge
- Upgrade to latest `amannn/action-semantic-pull-request` because of node.
- Upgrade to latest `marocchino/sticky-pull-request-comment` - node
- Upgrade to latest `derberg/npm-dependency-manager-for-your-github-org` - node
- Upgrade to latest `m1ner79/Github-Twittction` -node

### removal of one action

Remove `sentiment-analysis.yml` - it will not work because of node deprecation. The fact is that this action is not working for few months already, it was never good because of lack of context and was 100% of time causing false positives

### stop using `actions-ecosystem/action-create-comment`

Looks like `actions-ecosystem/action-create-comment` is not maintained and they did not bump version for node16. I removed it from `help-command.yml` and now use GitHub API directly in script action

### stop using `LoveToKnow/slackify-markdown-action`

Unfortunately `LoveToKnow/slackify-markdown-action` is not very much maintained and there is a PR to change to node16 for months, not merged. 

Now we do not use this action but for the first time we will use composite action -> https://github.com/asyncapi/.github/pull/226/files#diff-dacdba02324430e1c58a8c5262cf680d96abfc8ca95a5b3500cc9aa9ef10d155
I'm personally in love with composite actions and think we can optimize our CI with it a lot. **Important to notice** there composite actions do not have to be replicated with global workflow in multiple repos, can be referenced from this repo

this was tested in my test repo:
- test composite from my fork https://github.com/derberg/.github-asyncapi/blob/upgradeactions/.github/actions/slackify-markdown/action.yml
- test workflow using composite https://github.com/derberg/test-experiment/blob/master/.github/workflows/testtest.yml
- test comment that triggered workflow https://github.com/derberg/test-experiment/issues/149#issuecomment-1498607816
- test run https://github.com/derberg/test-experiment/actions/runs/4626483804/jobs/8183272577

### refactor `set-output`

`set-output` goes away -> https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

so they had to be refactored to a new way

### CODE REVIEW

**Comment by derberg** [comment]
cc @KhudaDad414 please review 🙏🏼 not an easy one

cc @14Richa would be cool if you could this time contribute as reviewer 🙏🏼 

cc @saurav9878 maybe you wanna have a look

**Review by KhudaDad414 — COMMENTED** [review]
LGTM! left some comments.

File: .github/workflows/release-announcements.yml [hunk 1]
```diff
@@ -14,11 +14,13 @@ jobs:
     name: Slack - notify on every release
     runs-on: ubuntu-latest
     steps:
+      - name: Checkout repository
```
[the remark this answers is not in the document]
Comment (derberg) [reply .github/workflows/release-announcements.yml:17]: yeah, they are actually not needed, good 🦅 👁️ 
checkout was needed as I could not figure out how to get these new types of actions from the remote repository, so idea was to replicate it and therefore checkout was needed to run them

**Review by KhudaDad414 — APPROVED** [review]
LGTM!

**Pull request merged**